### PR TITLE
worked on implementation of user profile

### PR DESCRIPTION
**Pull request by mohsinayaz**
#27

### CODE REVIEW

File: client/src/app/admin/profile/profile.component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  selector: 'em-profile',
+  templateUrl: './profile.component.html',
+  styleUrls: ['./profile.component.scss']
+})
+export class ProfileComponent implements OnInit {
+
+  constructor(private userService: UserService, private snackBar: MatSnackBar) { }
+  user: any;
+  editSection = false ;
+  EditInfoGroup: FormGroup;
```
Comment (AhsanAyaz) [inline client/src/app/admin/profile/profile.component.ts]: Rename `EditInfoGroup` to `editInfoForm`.
Always use `camelCase` for naming properties.

File: client/src/app/admin/profile/profile.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  closeEditSection() {
+    this.editSection = false;
+  }
+
+  edit() {
+   const formData = this.EditInfoGroup.value;
+   this.userService.updateUser(formData).subscribe(data => {
+      this.userService.setUser(formData);
+      this.getUser();
```
Comment (AhsanAyaz) [inline client/src/app/admin/profile/profile.component.ts]: Remove the `this.getUser` call from here.
There's no need to fetch the data back from server when we already have the updated values.
Just update the user object using new values on the front-end.

File: client/src/app/core/services/user/user.service.ts [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
 
   getUser(): any {
     const user  = JSON.parse(localStorage.getItem(EmanConstant.eman_user));
     return user;
   }
+  setUser(data): any {
+    localStorage.setItem(EmanConstant.eman_user, JSON.stringify(data));
+  }
+
+  updateUser(data): Observable<any> {
```
Comment (AhsanAyaz) [inline client/src/app/core/services/user/user.service.ts]: Create an interface of the user Object.
**DO NOT USE** `any`

File: client/src/app/core/services/user/user.service.ts [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
+    return this.http.put(`${this.apiBaseUrl}/user`, data)
+    .pipe(
+      map((res) => {
```
Comment (AhsanAyaz) [inline client/src/app/core/services/user/user.service.ts]: No need to use `map` if the `res` object is never changed.
Just do `return this.http.put(`${this.apiBaseUrl}/user`, data);`.
That's enough.

File: client/src/app/admin/profile/profile.component.ts (hunk 1 above)
Comment (mohsinayaz) [reply client/src/app/admin/profile/profile.component.ts]: done

File: client/src/app/admin/profile/profile.component.ts (hunk 2 above)
Comment (mohsinayaz) [reply client/src/app/admin/profile/profile.component.ts]: done

File: client/src/app/core/services/user/user.service.ts (hunk 3 above)
Comment (mohsinayaz) [reply client/src/app/core/services/user/user.service.ts]: done

File: client/src/app/core/services/user/user.service.ts (hunk 4 above)
Comment (mohsinayaz) [reply client/src/app/core/services/user/user.service.ts]: done

**Pull request merged**